src/main/java/io/beanmapper/strategy/MapToRecordStrategy.java

```diff
@@ -0,0 +1,158 @@
+package io.beanmapper.strategy;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.RecordComponent;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.beanmapper.BeanMapper;
+import io.beanmapper.annotations.RecordConstruct;
+import io.beanmapper.annotations.RecordConstructMode;
+import io.beanmapper.config.Configuration;
+import io.beanmapper.core.converter.BeanConverter;
+import io.beanmapper.exceptions.BeanInstantiationException;
+import io.beanmapper.exceptions.RecordConstructorConflictException;
+import io.beanmapper.utils.Records;
+
+/**
+ * MapToRecordStrategy offers a comprehensive implementation of the MapToClassStrategy, targeted towards mapping a class
+ * to a record.
+ *
+ * <p>A class (or record) can be mapped to a target record, simply by calling the map-method associated with this class.
+ * </p><p>Internally, the map-method will create dynamic classes as needed, to create suitable intermediaries. At least one,
+ * and at most two, intermediary classed can be generated. For class-generation, the
+ * {@link io.beanmapper.dynclass.ClassGenerator}-class is used.</p>
+ */
+public final class MapToRecordStrategy extends MapToClassStrategy {
+
+    MapToRecordStrategy(BeanMapper beanMapper, Configuration configuration) {
+        super(beanMapper, configuration);
+    }
+
+    @Override
+    public <S, T> T map(final S source) {
+
+        if (source.getClass().equals(this.getConfiguration().getTargetClass()))
```

Comment:
Je roept "this.getConfiguration().getTargetClass()" al vier keer aan in deze methode. Misschien wat makkelijker te lezen als je dit even naar een variable verwijst.